text/0004-new-pre-coordicide-consensus/0004-new-pre-coordicide-consensus.tex.md

```diff
@@ -0,0 +1,191 @@
+
+But first some definitions:
+
+`Tip`:
+A solid bundle tail of a bundle that has no approvers.
+
+`Approved transaction Roots`:
+All `seened` (by milestone) transaction that can be reached by walking from a given transaction down to its parents. The walk must terminate once we reached a `seened` transaction.
+
+`Transaction Snapshot Index`:
+The index of the milestone that marked the transaction as `seen`
+
+`Oldest Transaction Root Snapshot`:
+The milestone bundle with the lowest index that marked any of the Approved Transaction Roots as `seen`.
+
+`Youngest Transaction Root Snapshot`:
+The milestone bundle with the highest index that marked any of the Approved Transaction Roots as `seen`.
+
+
+
+### Timestamp based scoring 
+
+#### Configurable Values
+$C_1$ - Time in ms that a tip's timestamp can be *below* its solidification time.
+
+$C'_1$ - Time in ms that a tip's timestamp can be *above* its soldification time.
+
+$C_2$ - Max difference between tip solidification time and parent bundle solidification timestamp.
+
+$M$ - Max difference between tip solidification time and parent bundle signed timestamp.
+
+#### Definitions
+Let $t(x)$ be the signed timestamp and $r(x)$ the solidification time of transaction $x$. A tip will be marked as $v$ and its direct approved bundle tails are marked as $v_1$ and $v_2$. Let $M$ be some large constant. $otrs$ is `Oldest Transaction Root Snapshot`:
+
+#### Algorithm
+
+Score 0 will be given if one of the following is true:
+    
+1. $t(v)<r(v)-C_1$ or $t(v)>r(v)+C'_1$
+2. $r(v) - t(otrs(v))>M$
+3. All $v_i$ satisfy $r(v)-r(v_i) > C_2$
+4. if at least one $v_i$ has a score of 0 (to enforce monotonicity)
+
+Else Score 1 will be given if exactly one $v_i$ satisfies $r(v)-r(v_i) \leq C_2$
+
+Else Score 2 will be given.
+
+#### Recommended defaults
+
+$C_1$ - 30 seconds
+
+$C'_1$ - 30 seconds
+
+$C_2$ -  2 minutes
+
+$M$ - 15 minutes
+
+### Milestone based scoring
+
+#### Configurable Values
+$C_1$ - The threshold for approving transactions with a root snapshot index that is below the latest solid one.
+
+$C_2$ - Max difference between latest solid milestone index and parent $OTRSI$.
+
+$M$ - Max difference between latest solid milestone index and current transaction $OTRSI$. Current below max depth parameter.
+
+#### Definitions
+Let $ytrsi(v)$ be `Youngest Transaction Root Snapshot Index` for transaction $v$ 
+
+Let $otrsi(v)$ be `Oldest Transaction Root Snapshot Index` for transaction $v$
+
+Let $lsmi$ be `Last solid milestone index` 
+
+#### Algorithm
+
+Score 0  will be given if one of the following is true:
+ 
+1. $lsmi-ytrsi(v)>C_1$    
+2. $lsmi-otrsi(v)>M$    
+3. both $v_i$ satisfy $lsmi-otrsi(v_i)>C_2$
+4. at least one $v_i$ has a score of 0 (to enforce monotonicity)
+
+Else Score 1 will be given if exactly one $v_i$ satisfies $lsmi -otrsi(v_i) \leq C_2$
+
+Else Score 2 will given.
+
```

Comment:
Perhaps I'm missing something but shouldn't there be recommended defaults for C1,C2 and M for the milestone based scoring?

Reply:
oops:

C_1 - 2
C_2- 7
M - 15 minutes